glean-core/ffi/src/lib.rs

```diff
@@ -299,6 +300,117 @@ pub extern "C" fn glean_boolean_test_get_value(
     })
 }
 
+// *** Start of the Datetime FFI part ***
+
+#[no_mangle]
+pub extern "C" fn glean_new_datetime_metric(
+    category: FfiStr,
+    name: FfiStr,
+    send_in_pings: RawStringArray,
+    send_in_pings_len: i32,
+    lifetime: i32,
+    disabled: u8,
+    time_unit: i32,
+) -> u64 {
+    DATETIME_METRICS.insert_with_log(|| {
+        let send_in_pings = unsafe { from_raw_string_array(send_in_pings, send_in_pings_len) };
+        let lifetime = Lifetime::try_from(lifetime)?;
+        let tu = TimeUnit::try_from(time_unit)?;
+
+        Ok(DatetimeMetric::new(
+            CommonMetricData {
+                name: name.into_string(),
+                category: category.into_string(),
+                send_in_pings,
+                lifetime,
+                disabled: disabled != 0,
+            },
+            tu,
+        ))
+    })
+}
+
+#[no_mangle]
+pub extern "C" fn glean_datetime_set(
+    glean_handle: u64,
+    metric_id: u64,
+    year: i32,
+    month: u32,
+    day: u32,
+    hour: u32,
+    minute: u32,
+    second: u32,
+    nano: i64,
+    offset_seconds: i32,
+) {
+    use std::convert::TryInto;
+    // Convert and truncate the nanos to u32, as that's what the underlying
+    // library uses. Unfortunately, not all platform have unsigned integers
+    // so we need to work with what we have.
+    if nano < std::u32::MIN.try_into().unwrap() || nano > std::u32::MAX.try_into().unwrap() {
```

Comment:
The first part is just a very fancy way of saying `0`.
The second part is just `std::u32::MAX as i64` (as we [statically] upcast we don't have to worry about any wrappings))